### DISCUSSION TITLE
Keyboard Backlight Timeout Period

### DISCUSSION THREAD

**Discussion by Keitarono** [seerge/g-helper#332] [Ideas]
Not sure if this is possible to control through software or if it's all in the BIOS, but I think it would be cool to be able to choose how long the keyboard backlight stays on after you stop touching your laptop.

**Comment by seerge** [comment]
@Keitarono this topic was already raised https://github.com/seerge/g-helper/discussions/181 . Main reason i didn't add a setting for that , is cause it requires adjusting an entry in registry. And for that app will need admin permissions, that it doesn't use now. And it's seems too much to request admin rights just for one small addition :)

**Comment by steverand** [comment]
Before removing AC a year ago I had my keyboard set to never turn off but now it's reverted to 60 seconds. i've found the reg key and will change and reboot to test. Perhaps instead of admin access the app could read the value so the user knows what it is currently and perhaps the reg path to the key for manual configuration?

Thank you for such an awesome tool, I might make a post of what does and doesn't work on my laptop but the basics work and I am already thrilled!

**Reply by steverand** [reply]
That worked. Not sure why it reverted back but for anyone else, make .reg file or browse manually:

Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\ASUS\ASUS System Control Interface\AsusOptimization\ASUS Keyboard Hotkeys]
"TurnOffKeybdLight"=dword:00000000

**Reply by seerge** [reply]
[GHelper.zip](https://github.com/seerge/g-helper/files/11444318/GHelper.zip)
in this built i have added timeout control under Extra

**Comment by steverand** [comment]
Appreciated but the app doesn't run. It stays as a blue icon in system tray and then quits out. Compile issue or me?
-Steve

On Thu, May 11, 2023 at 02:48 AM, Serge  wrote: 
 GHelper.zip (https://github.com/seerge/g-helper/files/11444318/GHelper.zip)
 in this built i have added timeout control under Extra

**Reply by seerge** [reply]
@steverand based on 0 debug info that you provided, i can't answer :) Check event log in windows for crashes if you think it crashed.

**Reply by steverand** [reply]
Application: GHelper.exe
CoreCLR Version: 7.0.523.17405
.NET Version: 7.0.5
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException: Object reference not set to an instance of an object.
   at GHelper.SettingsForm.SetGPUClocks(Boolean launchAsAdmin) in C:\Users\serge\g-helper\app\Settings.cs:line 1059
   at GHelper.SettingsForm.SetPerformanceMode(Int32 PerformanceMode, Boolean notify) in C:\Users\serge\g-helper\app\Settings.cs:line 1237
   at GHelper.SettingsForm.AutoPerformance() in C:\Users\serge\g-helper\app\Settings.cs:line 1299
   at GHelper.Program.SetAutoModes() in C:\Users\serge\g-helper\app\Program.cs:line 145
   at GHelper.Program.Main(String[] args) in C:\Users\serge\g-helper\app\Program.cs:line 83


Faulting application name: GHelper.exe, version: 0.60.0.0, time stamp: 0x641de100
Faulting module name: KERNELBASE.dll, version: 10.0.22000.1880, time stamp: 0x099e63f3
Exception code: 0xe0434352
Fault offset: 0x0000000000052bac
Faulting process id: 0x2d58
Faulting application start time: 0x01d9838b1afe046a
Faulting application path: D:\Google Drive\tools\GHelper.exe
Faulting module path: C:\windows\System32\KERNELBASE.dll
Report Id: bc09de9e-9900-4a03-876c-899c94c66aa2
Faulting package full name: 
Faulting package-relative application ID: 



Not sure if appdata is pointing to the wrong spot. What is "C:\Users\serge\g-helper\app\Settings.cs" about?

**Reply by seerge** [reply]
[GHelper.zip](https://github.com/seerge/g-helper/files/11446455/GHelper.zip)
Thanks, Try this one

**Reply by steverand** [reply]
Perfect! It changes the registry value nicely thanks!
